src/hooks/useCloseOnEscape.tsx

```diff
@@ -0,0 +1,11 @@
+import { useEffect } from 'react'
+
+export function useCloseOnEscape(onClose?: () => void) {
```

Comment:
nit: This doesn't necessarily close - it just triggers a callback on Escape.

Could you rename the file and all uses to `useOnEscapeHandler(onEscape?: () => void) {`?